modules/cnvkit/batch/main.nf

```diff
@@ -62,28 +66,130 @@ process CNVKIT_BATCH {
     def target_args = targets ? "--targets $targets" : ""
     def reference_args = reference ? "--reference $reference" : ""
 
-    """
-    if $tumor_cram; then
-        samtools view -T $fasta $tumor -@ $task.cpus -o $tumor_out
-    fi
-    if $normal_cram; then
-        samtools view -T $fasta $normal -@ $task.cpus -o $normal_out
-    fi
-
-
-    cnvkit.py \\
-        batch \\
-        $tumor_out \\
-        $normal_args \\
-        $fasta_args \\
-        $reference_args \\
-        $target_args \\
-        --processes $task.cpus \\
-        $args
-
-    cat <<-END_VERSIONS > versions.yml
-    "${task.process}":
-        cnvkit: \$(cnvkit.py version | sed -e "s/cnvkit v//g")
-    END_VERSIONS
-    """
+    // somatic_mode cram_input
+    if (tumor_cram && normal_cram){
+        """
+        samtools view -T $fasta $fai_reference $tumor -@ $task.cpus -o $tumor_out
```

Comment:
you don't need the funky ##idx## thing?

Reply:
not if the `fasta_fai` is passed along with the `fasta` reference genome

Reply:
ok I'll ask about this on slack

Reply:
if both files are not local and are accessed instead via an https://, s3:// or other URL then the  "##idx##" would be needed.